diff -Naur tcl8.6.7/unix/configure tcl8.6.7-new/unix/configure
--- tcl8.6.7/unix/configure	2017-08-09 14:45:58.000000000 +0000
+++ tcl8.6.7-new/unix/configure	2019-04-02 00:57:03.077895061 +0000
@@ -6557,7 +6557,7 @@
     if test "$GCC" = yes; then
 
 	CFLAGS_OPTIMIZE=-O2
-	CFLAGS_WARNING="-Wall"
+	CFLAGS_WARNING=""
 
 else
 
diff -Naur tcl8.6.7/unix/tcl.m4 tcl8.6.7-new/unix/tcl.m4
--- tcl8.6.7/unix/tcl.m4	2017-08-09 14:45:58.000000000 +0000
+++ tcl8.6.7-new/unix/tcl.m4	2019-04-02 00:57:17.004318566 +0000
@@ -1095,7 +1095,7 @@
     CFLAGS_DEBUG=-g
     AS_IF([test "$GCC" = yes], [
 	CFLAGS_OPTIMIZE=-O2
-	CFLAGS_WARNING="-Wall"
+	CFLAGS_WARNING=""
     ], [
 	CFLAGS_OPTIMIZE=-O
 	CFLAGS_WARNING=""
